op-node/metrics/metrics.go

```diff
@@ -491,8 +500,25 @@ func (m *Metrics) RecordGossipEvent(evType int32) {
 	m.GossipEventsTotal.WithLabelValues(pb.TraceEvent_Type_name[evType]).Inc()
 }
 
+func getLabelForScore(score float64) string {
+	if score <= -40 {
+		return "banned"
+	} else if score <= 0 {
+		return "low"
+	} else if score <= 20 {
+		return "medium"
+	} else {
+		return "high"
+	}
+}
+
 func (m *Metrics) RecordPeerScoring(peerID peer.ID, score float64) {
-	m.PeerScores.WithLabelValues(peerID.String()).Set(score)
+	prevScore, ok := m.PeerStore[peerID]
+	if ok {
+		m.PeerScores.WithLabelValues(getLabelForScore(prevScore)).Dec()
+	}
+	m.PeerScores.WithLabelValues(getLabelForScore(score)).Inc()
+	m.PeerStore[peerID] = score
```

Comment:
Discussed offline with @refcell and @protolambda - there's a memory leak here because previous scores aren't removed when libp2p eventually forgets a peer ID.

Simplest solution is likely to do the grouping in `peer_scorer.go` by updating the `SnapshotHook` to something like:
```go
func (s *ScoreBands) Record(score float64) {
	if score <= -40 {
		s.Banned = s.Banned + 1
	} else if score <= 0 {
		s.Low = s.Low + 1
	} ... etc
}

// SnapshotHook returns a function that is called periodically by the pubsub library to inspect the peer scores.
// It is passed into the pubsub library as a [pubsub.ExtendedPeerScoreInspectFn] in the [pubsub.WithPeerScoreInspect] option.
// The returned [pubsub.ExtendedPeerScoreInspectFn] is called with a mapping of peer IDs to peer score snapshots.
func (s *scorer) SnapshotHook() pubsub.ExtendedPeerScoreInspectFn {
	return func(m map[peer.ID]*pubsub.PeerScoreSnapshot) {
		scoreBands := &ScoreBands{}
		for id, snap := range m {
			scoreBands.Record(snap.Score)

			// Update with the peer gater
			s.gater.Update(id, snap.Score)
		}
		s.metricer.RecordPeerScoring(scoreBands)
	}
}
```

The `ScoreBands` struct maybe should live in metrics module - not sure...